src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -504,9 +504,10 @@ parseNullBinder = underscore *> return NullBinder
 
 parseIdentifierAndBinder :: TokenParser (String, Binder)
 parseIdentifierAndBinder =
-    do name <- lname <|> stringLiteral
+    do name <- lname
        b <- P.option (VarBinder (Ident name)) rest
        return (name, b)
+    <|> (,) <$> stringLiteral <*> rest
```

Comment:
What's actually happening here in the first place? :smile: I don't quite understand what the `stringLiteral` is doing here.